### ISSUE TITLE
null is not an object (evaluating 'document.head.appendChild')

### ISSUE THREAD

**Issue by lifeiscontent** [remix-run/remix#8983]
### Reproduction

code in question exists here: https://github.com/remix-run/remix/blob/b6377ce479e26633afb8aade6e2ff51d756a7f0f/packages/remix-react/links.ts#L274-L289

### System Info

```shell
on Mobile Safari 15.2 I'm seeing in sentry logs:

null is not an object (evaluating 'document.head.appendChild')
```

### Used Package Manager

pnpm

### Expected Behavior

not have remix throw errors in prod

### Actual Behavior

remix throws errors in prod due to document.head being undefined/null

**Comment by brophdawg11** [comment]
I'm not sure what we can do here - can you provide any more information on the issue?  [`document.head`](https://developer.mozilla.org/en-US/docs/Web/API/Document/head) should be fine in every browser.  Is this specific to safari 15.2 in your case?  Are you rendering a `<head>` element?  Have you seen this happen in any other browsers?

**Comment by lifeiscontent** [comment]
@brophdawg11 not entirely sure, I've just seen these errors in sentry, I'm guessing you could conditionally check if the document.head exists before running these operations on it, I can send over the full log in sentry if you think that would help.

**Comment by lifeiscontent** [comment]
@brophdawg11 yeah, I agree with you, let me look into it more and see if I can provide a consistent reproduction
